build.Containerfile

```diff
@@ -0,0 +1,43 @@
+#
+# Reproducible reference build
+#
+# Usage:
+#
+# docker build --file build.Containerfile --output <outputdir> .
+# or
+# podman build --file build.Containerfile --output <outputdir> .
+#
+# The build artifacts are written to the specified output directory.
+#
+
+FROM debian:bullseye-slim AS build-stage
+
+# prepare debian environment
+ENV DEBIAN_FRONTEND noninteractive
+RUN /usr/bin/apt-get update && \
+    /usr/bin/apt-get --yes install openjdk-11-jdk-headless gradle && \
```

Comment:
This is potentially non-reproducible because the versions are not locked. In practice, however, I doubt it will be an issue.

Reply:
That's true. It's hard to strike the right balance between security patches and reproducibility. However I also don't expect much trouble here.

Sadly there is no Debian variant on Docker Hub that doesn't have `-updates` enabled.